### PR TITLE
SummaryNumber Component: Update to latest hifi design

### PR DESCRIPTION
**Pull request by ryelle**
See #131 - this PR adds a previous value/label to each SummaryNumber, and updates the design to match the updated hifi designs. This also adds in the ability to link off to a report by adding a required `href` prop to the SummaryNumber.

The smaller screen dropdown functionality will come in a later PR, as that will be much more complex.

**Demo on 1440px wide:**

large-screen-one
large-screen-two

**Demo on 1024px wide:**

med-screen-two

**Demo in a card**

screen shot 2018-07-25 at 11 09 03 am

**Demo of interactive states**

(top left is active/pressed, middle is selected, top right is hovered, bottom left is unselected + focused)
screen shot 2018-07-25 at 1 30 50 pm

(right is a selected + focused item)
screen shot 2018-07-25 at 1 31 04 pm

Don't be scared off by the `+676 lines`, 374 of that is creating an extensive demo page with lots of different number combos 😁

To test:

- Look at the example report `/wp-admin/admin.php?page=wc-admin#/analytics/test`
- Check that all SummaryNumbers appear as expected on screens larger than 783px (have not addressed the dropdowns yet)
- Specific "edge cases":
  - The 2nd item in most cases is "selected", the exception is in "Ten or More Data Points" where it's the 5th item to show selection on the second line.
  - The 3rd item in most cases has a delta of 0
  - In "Four Data Points", the 3rd item has no previous value or delta
  - In "Ten or More Data Points", the 8th item has a ridiculously large previous value
- Mouse or keyboard over them to check on the interactive states
- Check the same for the SummaryNumber card on the dashboard
- Try using VoiceOver to interact with the SummaryNumbers – I've called this a navigation menu, since they link off to specific reports/charts. You should be able to target and click on a summary number

**Note** All the summary numbers go to a default link of `/analytics`, because a) I don't know what these should link to, and b) I really didn't want to update every single demo `SummaryNumber` 😆

### CODE REVIEW

File: client/components/summary/style.scss [hunk 1]
```diff
@@ -1,72 +1,229 @@
 /** @format */
 
+// Set up some local color variables to make the CSS more clear
```
Comment (justinshreve) [inline client/components/summary/style.scss]: 👍

File: client/components/summary/README.md [hunk 2]
```diff
@@ -28,6 +28,9 @@ render: function() {
 
 * `label` (required): A string description of this value, ex "Revenue", or "New Customers"
 * `value` (required): A string or number value to display - a string is allowed so we can accept currency formatting.
+* `href` (required): An internal link to the report focused on this number.
```
Comment (justinshreve) [inline client/components/summary/README.md]: Minor: The README example should include `href` since it is a required prop.

Should it also mention that '/analytics' is default, from defaultProps, Or is that just temporary for these examples?

**Review by emdashcodes — APPROVED** [review]
This is looking great! I have one little minor comment about the README, but otherwise nothing is jumping out at me.

I do have one question about the 'selected' style vs focus, but It's more of a design question and definitely not a blocker for this PR.. 
How does a number get 'selected' with the purple treatment? If these are just links, seems like you would only need a focus style and that the focus style should be the one with the purple border and white background?

File: client/components/summary/README.md (hunk 2 above)
Comment (ryelle) [reply client/components/summary/README.md]: I intended for the /analytics default to be temporary, but yeah the readme example should get updated.

**Comment by ryelle** [comment]
> How does a number get 'selected' with the purple treatment?

I assumed it was the "currently displaying data point", so it's based on a prop set by the parent component (the report). In the designs it's called "active" state, but I'm pretty sure it's meant to be this selected state, and "pressed" is the design for `:active`. I could also use a designer's 👍 / 👎 to confirm though 🙂

File: client/analytics/report/example.js [hunk 3]
```diff
@@ -21,20 +21,388 @@ export default class extends Component {
...
 					<SummaryNumber
+						label={ __( 'Gross Revenue', 'wc-admin' ) }
+						value={ '$829.40' }
+						prevValue={ '$785.90' }
+						delta={ 5.5 }
+						selected
+					/>
+				</SummaryList>
+
+				<h2>Two Data Points</h2>
```
Comment (timmyc) [inline client/analytics/report/example.js]: So. Many. Data. Points.

File: client/components/summary/index.js [hunk 4]
```diff
@@ -14,10 +16,23 @@ const SummaryList = ( { children, label } ) => {
 	if ( ! label ) {
 		label = __( 'Performance Indicators', 'wc-admin' );
 	}
+	// We default to "one" because we can't have empty children. If `children` is just one item,
+	// it's not an array and .length is undefined.
+	let hasItemsClass = 'has-one-item';
+	if ( children && children.length ) {
+		const length = children.filter( Boolean ).length;
```
Comment (timmyc) [inline client/components/summary/index.js]: Very clever way to determine real child elements here, nice!

File: client/dashboard/store-performance/index.js [hunk 5]
```diff
@@ -97,7 +110,7 @@ class StorePerformance extends Component {
 
 export default compose( [
 	withAPIData( () => ( {
-		orders: '/wc/v2/orders?status=processing',
-		products: '/wc/v2/products',
+		orders: '/wc/v3/orders?status=processing',
```
Comment (timmyc) [inline client/dashboard/store-performance/index.js]: 🚀

**Review by timmyc — APPROVED** [review]
This is awesomeness!! Testing out great for me - and I enjoyed attending a clinic by reading the css in this PR.

**Comment by LevinMedia** [comment]
This is looking great! 

>I've called this a navigation menu, since they link off to specific reports/charts.

This is partly the case: Summary numbers on the dashboard (/admin.php?page=wc-admin#/) will link to different reports. When a summary number is selected in a report, (the purple highlight treatment) the data in the graph updates to reflect data in selected summary number. 

>Try using VoiceOver to interact with the SummaryNumbers – I've called this a navigation menu, since they link off to specific reports/charts.

This may not be the best approach given the pattern I mentioned above - When focusing to a new summary number, VoiceOver says it is a link, which isn't actually the case. In most cases the summary numbrers act more like a radio button, selecting what data is displayed directly below in the graph. The exception are summary numbers on the dashboard, as they link to their respective reports. 

> I'm pretty sure it's meant to be this selected state, and "pressed" is the design for :active

You've got that correct, thank you! 

A few additional notes:

Interior borders should be using light grey 500 (#E2E4E7), whereas the outer border that wraps the component should be using light grey 700 (#CCD0D4).

The text styles for the `woocommerce-summary__item-value` and `woocommerce-summary__item-delta-value` should change depending on the state of the summary number. They're bolder for the selected state of the component.

image

Any time a component width is <250px, the previous period / year value should always wrap to a new line, even if the first line has enough space to accomodate the value: 

screen shot 2018-07-25 at 11 31 01 pm

**Comment by ryelle** [comment]
> When a summary number is selected in a report, (the purple highlight treatment) the data in the graph updates to reflect data in selected summary number.

Even so, if it changes the URL (like with a query string), it should be a link. It'll also be more consistent behavior through the app from dash to reports. Functionally it'll do exactly what you're asking, it's just the semantics for VoiceOver.

> Interior borders should be using light grey 500 (#E2E4E7), whereas the outer border that wraps the component should be using light grey 700 (#CCD0D4).

~They should be, are you seeing something different?~ They _were_ but when I switched them to links the CSS broke 😩 

> Any time a component width is <250px, the previous period / year value should always wrap to a new line, even if the first line has enough space to accomodate the value:

We can't do responsive styles at the component level like that, but I can follow the same logic as for wrapping the percentage value.

**Comment by ryelle** [comment]
I've updated the font weights, previous value wrapping, and border colors. For the border colors we can only really change the horizontal borders, the vertical ones are trickier because we can't easily rely on nth-of-type patterns for changing the bottom row. I can revisit this in a later PR if we need to.

screen shot 2018-07-26 at 10 14 58 am

**Review by LevinMedia — APPROVED** [review]
Looking great! Thanks for making the previous update suggestions.
